[CI] SearchServiceSingleNodeTests testSlicingBehaviourForParallelCollection failing

**Build Scans:**
- [elasticsearch-periodic #7356 / release-tests](https://gradle-enterprise.elastic.co/s/lgvspdpqb35de)
- [elasticsearch-pull-request #65610 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/xvn47qo7neiey)
- [elasticsearch-pull-request #65594 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/deykvqzzqrmmk)
- [elasticsearch-pull-request #65602 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/qhbifprfks2o4)
- [elasticsearch-pull-request #65594 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/zwl6wrxxiuda4)
- [elasticsearch-periodic #7349 / release-tests](https://gradle-enterprise.elastic.co/s/xajajthyth72k)
- [elasticsearch-periodic #7342 / release-tests](https://gradle-enterprise.elastic.co/s/founp2av4d2jo)
- [elasticsearch-periodic #7328 / release-tests](https://gradle-enterprise.elastic.co/s/rqyap3ryiab2s)
- [elasticsearch-periodic #7321 / release-tests](https://gradle-enterprise.elastic.co/s/qjklqxpqwdoza)
- [elasticsearch-periodic #7314 / release-tests](https://gradle-enterprise.elastic.co/s/abpt6b7mtyqcc)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.search.SearchServiceSingleNodeTests.testSlicingBehaviourForParallelCollection" -Dtests.seed=A5774D4729EE9273 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=szl -Dtests.timezone=PST8PDT -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.search.SearchServiceSingleNodeTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testSlicingBehaviourForParallelCollection),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: null
```

**Issue Reasons:**
- [main] 6 consecutive failures in step release-tests
- [main] 4 consecutive failures in step checkpart1_release-tests
- [main] 10 failures in test testSlicingBehaviourForParallelCollection (3.3% fail rate in 304 executions)
- [main] 6 failures in step release-tests (100.0% fail rate in 6 executions)
- [main] 4 failures in step checkpart1_release-tests (100.0% fail rate in 4 executions)
- [main] 6 failures in pipeline elasticsearch-periodic (85.7% fail rate in 7 executions)
- [main] 3 failures in pipeline elasticsearch-pull-request (3.7% fail rate in 81 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -345,9 +345,6 @@ tests:
 - class: org.elasticsearch.xpack.test.rest.XPackRestIT
   method: test {p0=transform/transforms_stats/Test get transform stats with timeout}
   issue: https://github.com/elastic/elasticsearch/issues/125975
-- class: org.elasticsearch.search.SearchServiceSingleNodeTests
-  method: testSlicingBehaviourForParallelCollection
-  issue: https://github.com/elastic/elasticsearch/issues/125899
 - class: org.elasticsearch.packaging.test.DockerTests
   method: test021InstallPlugin
   issue: https://github.com/elastic/elasticsearch/issues/116147

--- a/server/src/main/java/org/elasticsearch/search/SearchService.java
+++ b/server/src/main/java/org/elasticsearch/search/SearchService.java
@@ -416,12 +416,12 @@ public class SearchService extends AbstractLifecycleComponent implements IndexEv
 
         enableQueryPhaseParallelCollection = QUERY_PHASE_PARALLEL_COLLECTION_ENABLED.get(settings);
         if (BATCHED_QUERY_PHASE_FEATURE_FLAG) {
-            clusterService.getClusterSettings()
-                .addSettingsUpdateConsumer(QUERY_PHASE_PARALLEL_COLLECTION_ENABLED, this::setEnableQueryPhaseParallelCollection);
             batchQueryPhase = BATCHED_QUERY_PHASE.get(settings);
         } else {
             batchQueryPhase = false;
         }
+        clusterService.getClusterSettings()
+            .addSettingsUpdateConsumer(QUERY_PHASE_PARALLEL_COLLECTION_ENABLED, this::setEnableQueryPhaseParallelCollection);
         clusterService.getClusterSettings()
             .addSettingsUpdateConsumer(BATCHED_QUERY_PHASE, bulkExecuteQueryPhase -> this.batchQueryPhase = bulkExecuteQueryPhase);
         memoryAccountingBufferSize = MEMORY_ACCOUNTING_BUFFER_SIZE.get(settings).getBytes();

